build.gradle

```
@@ -341,7 +341,7 @@ subprojects {
                 artifact task
             }
 
-            artifactId = archivesBaseName
+            artifactId = project.extensions.findByType(BasePluginExtension)?.archivesName?.get()
```

I think this has the same effect, and looks a bit better. The "safe navigation operator" `?` also makes me think that this could be null, when it really should never be null, and the publishing plugin can't handle a null.
```suggestion
            artifactId = base.archivesName.get()
```

---

Cool, this makes sense, I will modify it.